### ISSUE TITLE
Updates to Gallery block not saved because of - 400 failed to load resource error - if more than 100 images are in gallery

### ISSUE THREAD

**Issue by wddean** [WordPress/gutenberg#46088]
If over 100 images are added to a Gallery block the "Image Size" setting never finishes loading ( the loading indicator spins indefinately and reads "Loading options...".

DevTools console shows the following error:
"Failed to load resource: the server responded with a status of 400 ()"

{"code":"rest_invalid_param","message":"Invalid parameter(s): per_page","data":{"status":400,"params":{"per_page":"per_page must be between 1 (inclusive) and 100 (inclusive)"},"details":{"per_page":{"code":"rest_out_of_bounds","message":"per_page must be between 1 (inclusive) and 100 (inclusive)","data":null}}}}

If a change is made to the "Link To" setting for the gallery this update is not saved. For example if "Link To" is updated to "Media File" images are not linked to their media file.

The error goes away and the "Image Size" setting finishes spinning and loads if there are 100 or fewer images in the gallery but just clicking update to apply the "LInk To" setting doesn't work. You first must change it to something else such as "Attachment Page" and then change it back to "Media File" and click update and then the Media file links get applied to all images.

**Comment by glendaviesnz** [comment]
I was able to replicate this by adding a gallery with > 100 images. @SiobhyB this is caused [by this change](https://github.com/WordPress/gutenberg/pull/40947). With this change the call to get the media info fails if a page size of > 100 is set. @Mamaduka any thoughts on ways around this, as the previous alternative of `-1` also [caused some concerns](https://github.com/WordPress/gutenberg/pull/34389#issuecomment-938550798)?

**Comment by Mamaduka** [comment]
It's okay to use an unbound query for web versions. The results are limited by the `include` parameter.

**Comment by SiobhyB** [comment]
Thank you for the ping @glendaviesnz! As a native-specific version of the `use-get-media.js` file has been introduced since I made that change, the unbounded query can be restored to the web file (which I see has already been proposed in https://github.com/WordPress/gutenberg/pull/46143) without impacting native. I'll also do some testing to see whether the issue with saving the Gallery block can be replicated on native and follow up accordingly. Appreciate the heads up 🙇‍♀️

**Comment by SiobhyB** [comment]
To tie up any loose ends: I finished testing this on native and am happy to report that the issues with 100+ images aren't reproducible there. I tested on both platforms with posts that have 100+ images uploaded across multiple blocks as well as posts that have a single gallery block with 100+ images. I was still able to save and make changes in those posts, without error, though the performance was noticeably slow. I don't think there are any further native-specific issues that need to be created here.
